Podfile

```diff
@@ -232,7 +232,7 @@ abstract_target 'Apps' do
 
   pod 'Gridicons', '~> 1.1.0'
 
-  pod 'WordPressAuthenticator', '~> 5.1-beta'
+  pod 'WordPressAuthenticator', '~> 5.6-beta'
```

Comment:
Before merging the PR, probably we should check if there are any important changes between `5.1` to `5.6` in `WordPressAuthenticator`.

Reply:
Good point. 

The more interesting file to look at is `Podfile.lock` where WordPressAuthenticator results resolved as version 5.5.0.

The only differences between [5.5.0](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/releases/tag/5.5.0) and [5.6.0-beta.1](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/releases/tag/5.6.0-beta.1) are of course all and only the changes bundled with 5.6.0-beta.1, and those are nothing more that the WordPressKit major version bump, which this PR verified.

I think this version requirement change is safe to merge. Even if an issue was introduced in going from `~> 5.1-beta` to `~> 5.6-beta` the culprit would not be in 5.6.0-beta.1

Reply:
By the way, was there a requirement in the code for something that came in WordPressAuthenticator 5.6.0?

Otherwise, we could have left the requirement as it was and trust in CocoaPods to resolve the correct WordPressAuthenticator version compatible with WordPressKit 7.0.0-beta.1.

Out of curiosity, I reverted this change on my machine and everything worked fine.

No big deal, though. We'll soon need to actually updated WordPressAuthenticator anyway: #20128